pkg/certs/certs.go

```diff
@@ -214,8 +214,12 @@ func GenerateSecret(name string, subject string, hosts string, ca *corev1.Secret
 }
 
 func GenerateSecretWithExpiration(name string, subject string, hosts string, expiration time.Duration, ca *corev1.Secret) corev1.Secret {
-	caCert := getCAFromSecret(ca)
-	return generateSecret(name, subject, hosts, &caCert, expiration)
+	var caCert *CertificateAuthority
```

Comment:
Minor issue: feels to me like it would be nicer to change getCAFromSecret to return nil if the secret is nil or does not contain a valid CA.

Reply:
I agree @grs. It is better to transfer the responsibility to `getCAFromSecret` instead.
Done within the last commit.